# Instalação e importação das bibliotecas

Instalação das bibliotecas se necessário

In [2]:
!pip install numpy pandas matplotlib seaborn 

Import bibliotecas com alias convencionais.

In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


# EDA

## Carregando dados
Utilização da biblioteca pandas para leitura dos dados

In [4]:
df_salarios = pd.read_csv(r'./Dados/Salarios_dep_carg_query_1.csv')
df_media_salario = pd.read_csv(r'./Dados/Media_salarios_dep_carg.csv')
df_salario_localizacao = pd.read_csv(r'./Dados/Salarios_localizacao_query_2.csv')
df_media_salario_localizacao = pd.read_csv(r'./Dados/Media_salarios_localizacao.csv')

Prévisualização dos dados para conferencia da importação correta

In [5]:
df_salarios.head(10)

,DEPARTAMENTO,CARGOS,SALARIO
0,Executive,President,24000
1,Executive,Administration Vice President,17000
2,Executive,Administration Vice President,17000
3,Sales,Sales Manager,14000
4,Sales,Sales Manager,13500
5,Marketing,Marketing Manager,13000
6,Finance,Finance Manager,12008
7,Accounting,Accounting Manager,12008
8,Sales,Sales Manager,12000
9,Sales,Sales Representative,11500


In [6]:
df_media_salario

,DEPARTAMENTO,CARGOS,MÉDIA_SALARIAL
0,Executive,President,24000.000000
1,Executive,Administration Vice President,17000.000000
2,Marketing,Marketing Manager,13000.000000
3,Sales,Sales Manager,12200.000000
4,Accounting,Accounting Manager,12008.000000
5,Finance,Finance Manager,12008.000000
6,Purchasing,Purchasing Manager,11000.000000
7,Public Relations,Public Relations Representative,10000.000000
8,Sales,Sales Representative,8396.551724
9,Accounting,Public Accountant,8300.000000


In [ ]:
df_salario_localizacao.head(10)


,DEPARTAMENTO,CARGOS,CIDADE,PAIS,REGIAO,SALARIO
0,Executive,President,Seattle,United States of America,Americas,24000
1,Executive,Administration Vice President,Seattle,United States of America,Americas,17000
2,Executive,Administration Vice President,Seattle,United States of America,Americas,17000
3,Sales,Sales Manager,Oxford,United Kingdom of Great Britain and Northern I...,Europe,14000
4,Sales,Sales Manager,Oxford,United Kingdom of Great Britain and Northern I...,Europe,13500
5,Marketing,Marketing Manager,Toronto,Canada,Americas,13000
6,Finance,Finance Manager,Seattle,United States of America,Americas,12008
7,Accounting,Accounting Manager,Seattle,United States of America,Americas,12008
8,Sales,Sales Manager,Oxford,United Kingdom of Great Britain and Northern I...,Europe,12000
9,Sales,Sales Representative,Oxford,United Kingdom of Great Britain and Northern I...,Europe,11500


In [33]:
df_media_salario_localizacao

,DEPARTAMENTO,CARGOS,CIDADE,PAIS,REGIAO,MEDIA_SALARIAL
0,Executive,President,Seattle,United States of America,Americas,24000.00
1,Executive,Administration Vice President,Seattle,United States of America,Americas,17000.00
2,Marketing,Marketing Manager,Toronto,Canada,Americas,13000.00
3,Sales,Sales Manager,Oxford,United Kingdom of Great Britain and Northern I...,Europe,12200.00
4,Accounting,Accounting Manager,Seattle,United States of America,Americas,12008.00
5,Finance,Finance Manager,Seattle,United States of America,Americas,12008.00
6,Purchasing,Purchasing Manager,Seattle,United States of America,Americas,11000.00
7,Public Relations,Public Relations Representative,Munich,Germany,Europe,10000.00
8,Sales,Sales Representative,Oxford,United Kingdom of Great Britain and Northern I...,Europe,8396.55
9,Accounting,Public Accountant,Seattle,United States of America,Americas,8300.00


## ADA

### df_salarios

In [ ]:

df_salarios.info()
df_salarios.value_counts()


<class 'pandas.DataFrame'>
RangeIndex: 106 entries, 0 to 105
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   DEPARTAMENTO  106 non-null    str  
 1   CARGOS        106 non-null    str  
 2   SALARIO       106 non-null    int64
dtypes: int64(1), str(2)
memory usage: 2.6 KB


DEPARTAMENTO  CARGOS                         SALARIO
Sales         Sales Representative           10000      3
                                             9500       3
Shipping      Stock Clerk                    2500       3
Executive     Administration Vice President  17000      2
Sales         Sales Representative           9000       2
                                                       ..
Shipping      Stock Clerk                    2800       1
Purchasing    Purchasing Clerk               2600       1
Shipping      Stock Clerk                    2600       1
Purchasing    Purchasing Clerk               2500       1
Shipping      Stock Clerk                    2100       1
Name: count, Length: 82, dtype: int64

Podemos ver que não há nulos, existem 2 dados categóricos e um numérico, há mais cargos em vendas que em outros departamentos

In [43]:
print(df_salarios["CARGOS"].value_counts())
print("-"*35)
print(df_salarios["DEPARTAMENTO"].value_counts())
print("-"*35)
print(df_salarios[["CARGOS", "DEPARTAMENTO"]].value_counts())

CARGOS
Sales Representative               29
Shipping Clerk                     20
Stock Clerk                        20
Sales Manager                       5
Programmer                          5
Accountant                          5
Stock Manager                       5
Purchasing Clerk                    5
Administration Vice President       2
President                           1
Marketing Manager                   1
Finance Manager                     1
Accounting Manager                  1
Purchasing Manager                  1
Public Relations Representative     1
Public Accountant                   1
Human Resources Representative      1
Marketing Representative            1
Administration Assistant            1
Name: count, dtype: int64
-----------------------------------
DEPARTAMENTO
Shipping            45
Sales               34
Finance              6
Purchasing           6
IT                   5
Executive            3
Marketing            2
Accounting           2
Public Relat

Podemos ver que o departamento de Shiping é o maior, seguido pelo de vendas

In [29]:
print(df_salarios.describe())
Q1 = df_salarios["SALARIO"].quantile(0.25)
Q3 = df_salarios["SALARIO"].quantile(0.75)
IQR = Q3 - Q1
print(f"O IQR é de {IQR}")

            SALARIO
count    106.000000
mean    6456.754717
std     3927.798234
min     2100.000000
25%     3100.000000
50%     6150.000000
75%     8950.000000
max    24000.000000
O IQR é de 5850.0


**Analise da coluna salario:**
 Média e mediana estão próximas, porém o desbio padrão é bem grande, demonstrando uma grande diferença entre os valores, podemos já notar que há outliers já que o valor max é de 24000 enquanto 75% dos dados estão até 8950.

In [45]:
limite_superior = Q3 + (1.5 * IQR)
outliers = df_salarios[df_salarios["SALARIO"] > limite_superior]
print(outliers)

  DEPARTAMENTO     CARGOS  SALARIO
0    Executive  President    24000


Podemos ver que o unico oulier é o presidente, o que faz sentido

In [22]:
# Analisando duplicadas
print(f"Total de linhas duplicadas: {df_salarios.duplicated().sum()}")
duplicadas = df_salarios[df_salarios.duplicated()]
print(duplicadas)

Total de linhas duplicadas: 24
    DEPARTAMENTO                         CARGOS  SALARIO
2      Executive  Administration Vice President    17000
17         Sales           Sales Representative    10000
18         Sales           Sales Representative    10000
21         Sales           Sales Representative     9500
22         Sales           Sales Representative     9500
24         Sales           Sales Representative     9000
35         Sales           Sales Representative     8000
40         Sales           Sales Representative     7500
45         Sales           Sales Representative     7000
52         Sales           Sales Representative     6200
58            IT                     Programmer     4800
71      Shipping                    Stock Clerk     3300
74      Shipping                    Stock Clerk     3200
75      Shipping                 Shipping Clerk     3200
79      Shipping                 Shipping Clerk     3100
81      Shipping                 Shipping Clerk     3000


As duplicadas são cargos iguais que ganham o mesmo valor, o que faz sentido inicialmente, assim manteremos elas.

### df_salario_localizacao

In [34]:
df_salario_localizacao.info()
df_salario_localizacao.value_counts()

<class 'pandas.DataFrame'>
RangeIndex: 106 entries, 0 to 105
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   DEPARTAMENTO  106 non-null    str  
 1   CARGOS        106 non-null    str  
 2   CIDADE        106 non-null    str  
 3   PAIS          106 non-null    str  
 4   REGIAO        106 non-null    str  
 5   SALARIO       106 non-null    int64
dtypes: int64(1), str(5)
memory usage: 5.1 KB


DEPARTAMENTO  CARGOS                         CIDADE               PAIS                                                  REGIAO    SALARIO
Sales         Sales Representative           Oxford               United Kingdom of Great Britain and Northern Ireland  Europe    10000      3
                                                                                                                                  9500       3
Shipping      Stock Clerk                    South San Francisco  United States of America                              Americas  2500       3
Executive     Administration Vice President  Seattle              United States of America                              Americas  17000      2
Sales         Sales Representative           Oxford               United Kingdom of Great Britain and Northern Ireland  Europe    9000       2
                                                                                                                                            ..
Ship

Podemos ver que não há nulos, existem 5 dados categóricos e 1 numérico, há mais cargos em vendas de oxford que em outros departamentos

Como a coluna salário é a mesma não há necessidade de analisarmos novamente.

In [50]:
print(df_salario_localizacao["CIDADE"].value_counts())
print("-"*35)
print(df_salario_localizacao["PAIS"].value_counts())
print("-"*35)
print(df_salario_localizacao["REGIAO"].value_counts())
print("-"*35)
print(df_salario_localizacao[["CIDADE", "PAIS", "REGIAO"]].value_counts())

CIDADE
South San Francisco    45
Oxford                 34
Seattle                18
Southlake               5
Toronto                 2
Munich                  1
London                  1
Name: count, dtype: int64
-----------------------------------
PAIS
United States of America                                68
United Kingdom of Great Britain and Northern Ireland    35
Canada                                                   2
Germany                                                  1
Name: count, dtype: int64
-----------------------------------
REGIAO
Americas    70
Europe      36
Name: count, dtype: int64
-----------------------------------
CIDADE               PAIS                                                  REGIAO  
South San Francisco  United States of America                              Americas    45
Oxford               United Kingdom of Great Britain and Northern Ireland  Europe      34
Seattle              United States of America                              Americas

Podemos ver que mais da metade dos funcionarios da empresa são dos EUA, sendo maior parte de South San Francisco.

In [36]:
# Analisando duplicadas
print(f"Total de linhas duplicadas: {df_salario_localizacao.duplicated().sum()}")
duplicadas = df_salario_localizacao[df_salario_localizacao.duplicated()]
print(duplicadas)

Total de linhas duplicadas: 24
    DEPARTAMENTO                         CARGOS               CIDADE  \
2      Executive  Administration Vice President              Seattle   
16         Sales           Sales Representative               Oxford   
18         Sales           Sales Representative               Oxford   
21         Sales           Sales Representative               Oxford   
22         Sales           Sales Representative               Oxford   
25         Sales           Sales Representative               Oxford   
35         Sales           Sales Representative               Oxford   
40         Sales           Sales Representative               Oxford   
45         Sales           Sales Representative               Oxford   
52         Sales           Sales Representative               Oxford   
58            IT                     Programmer            Southlake   
71      Shipping                    Stock Clerk  South San Francisco   
74      Shipping                 

Corroborando a analise anterior há dados duplicados, esses em sua maioria são de vendas de oxford, south san Francisco e EUA